include/hal/library/responder/csrlib.h

```diff
@@ -48,6 +48,59 @@ extern bool libspdm_gen_csr(uint32_t base_hash_algo, uint32_t base_asym_algo, bo
                             bool is_device_cert_model);
+
+/**
+ * Gen CSR, which is uesd for SPDM 1.3
+ *
+ *   If the device need reset to gen csr, the all case is in the table.
+ *   | Overwrite | Req CSRTtrackingTag | Pending CSR | Reset |             Res Action       |
+ *   |-----------|---------------------|-------------|-------|------------------------------|
+ *   |    No     |          0          |      No     |   -   |          ResetRequired       |
+ *   |    No     |          0          |      Yes    |   -   |     ResetRequired or Busy    |
+ *   |    No     |        Non-0        |   No Match  |   -   |           Unexpected         |
+ *   |    No     |        Non-0        |     Match   | Before|             Busy             |
+ *   |    No     |        Non-0        |     Match   | After |             CSR              |
+ *   |    Yes    |          0          |      No     |   -   |          ResetRequired       |
+ *   |    Yes    |          0          |      Yes    |   -   |          ResetRequired       |
+ *   |    Yes    |        Non-0        |      -      |   -   |             Invalid          |
+ *
+ * @param[in]      base_hash_algo        Indicates the hash algorithm.
+ * @param[in]      base_asym_algo        Indicates the signing algorithm.
+ * @param[in]      need_reset            If true, then device needs to be reset to complete the CSR.
+ *                                       If false, the device doesn`t need to be reset to complete the CSR.
+ *
+ * @param[in]      request                A pointer to the SPDM request data.
+ * @param[in]      request_size           The size of SPDM request data.
+ *
+ * @param[in]      requester_info         Requester info to generate the CSR.
+ * @param[in]      requester_info_length  The length of requester info.
+ *
+ * @param[in]      opaque_data            opaque data to generate the CSR.
+ * @param[in]      opaque_data_length     The length of opaque data.
+ *
+ * @param[in]      csr_len      For input, csr_len is the size of store CSR buffer.
+ *                              For output, csr_len is CSR len for DER format
+ * @param[in]      csr_pointer  On input, csr_pointer is buffer address to store CSR.
+ *                              On output, csr_pointer is address for stored CSR.
+ *                              The csr_pointer address will be changed.
+ *
+ * @param[in]       req_cert_model        indicates the desired certificate model of the CSR
+ *
+ * @param[in]       req_csr_tracking_tag  this field shall contain the CSRTrackingTag of the associated GET_CSR request.
+ * @param[in]       overwrite             If set, the Responder shall stop processing any existing GET_CSR request and
+ *                                        overwrite it with this request
+ *
+ * @retval  true   Success.
+ * @retval  false  Failed to gen CSR.
+ **/
+extern bool libspdm_gen_csr_ex(uint32_t base_hash_algo, uint32_t base_asym_algo, bool *need_reset,
```

Comment:
Requiring this function is a breaking change and will need a new macro `LIBSPDM_ENABLE_CAPABILITY_CSR_CAP_EX`. Feel free to put it in `libspdm_lib_config.h` and specify that is only needed if the endpoints supports SPDM 1.3.

Reply:
I have added the L`IBSPDM_ENABLE_CAPABILITY_CSR_CAP_EX`.  Both `libspdm_gen_csr_ex` and `libspdm_get_csr_ex` are under this MACRO.

Reply:
Finally, add `LIBSPDM_ENABLE_CAPABILITY_CSR_CAP_EX` to https://github.com/DMTF/libspdm/blob/53f1cf915ed7cfd40b6712c722e8ef140cd7606a/.github/workflows/build.yml#L50-L51.